sdk/core/core-client-rest/test/sendRequest.spec.ts

```diff
@@ -83,6 +83,29 @@ describe("sendRequest", () => {
         contentType: "multipart/form-data",
       });
     });
+
+    it("should handle request body as FormData with array of binary", async () => {
+      const expectedFormData = { fileName: "foo.txt" };
+      const mockPipeline: Pipeline = createEmptyPipeline();
```

Comment:
Could you help me understand below case?

Imagine we have two arrays of files

```typescript
const array1 = [{name: "f1", content: {...}}, {name: "f2", content: {...}}];
const array2 = [{name: "f3", content: {...}}, {name: "f4", content: {...}}];
```

What does the http payload look like?

Reply:
I notice the cadl-ranch case [here](https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/payload/multipart/main.tsp#L119-L129). And the payload would be like that I think.

Reply:
Yes, it would look like that once core-rest-pipeline has done its thing.